django/contrib/auth/views.py

```diff
@@ -55,21 +55,28 @@ def inner(*args, **kwargs):
 def login(request, template_name='registration/login.html',
           redirect_field_name=REDIRECT_FIELD_NAME,
           authentication_form=AuthenticationForm,
-          extra_context=None):
+          extra_context=None, redirect_authenticated_user=False):
     """
     Displays the login form and handles the login action.
     """
     redirect_to = request.POST.get(redirect_field_name,
                                    request.GET.get(redirect_field_name, ''))
 
-    if request.method == "POST":
-        form = authentication_form(request, data=request.POST)
-        if form.is_valid():
+    # Ensure the user-originating redirection url is safe.
+    if not is_safe_url(url=redirect_to, host=request.get_host()):
```

Comment:
Would it be worth using an inner function named something like `get_redirect_url()` to avoid this logic when it's not needed?